src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/location/importer/LocationsImporter.kt

```diff
@@ -35,16 +36,21 @@ class LocationsImporter(private val repo: LocationRepository,
 
             val rows = sheet.drop(1).filterNot { it.getCell(1)?.stringCellValue.isNullOrBlank() }
 
-            rows.forEach { row ->
-                total++
-                Result.runCatching { repo.save(locationType.toLocation(row)) }
-                        .onFailure { errors.add(it.message + " " + it.cause.toString()) }
+            rows.forEachIndexed{ index, row ->
```

Comment:
maybe make errors immutable and use `map.filter()` rather than forEach to return the errors?